service/worker/scheduler/workflow_test.go

```diff
@@ -345,6 +361,7 @@ func (s *workflowSuite) TestStart() {
 		s.Equal(`"value"`, payload.ToString(req.Request.SearchAttributes.IndexedFields["myfield"]))
 		s.Equal(`"myschedule"`, payload.ToString(req.Request.SearchAttributes.IndexedFields[searchattribute.TemporalScheduledById]))
 		s.Equal(`"2022-06-01T00:15:00Z"`, payload.ToString(req.Request.SearchAttributes.IndexedFields[searchattribute.TemporalScheduledStartTime]))
+		s.Equal(userMetadata, req.Request.GetUserMetadata())
```

Comment:
I think you should use [protoassert.ProtoEqual](https://github.com/temporalio/temporal/blob/main/common/testing/protoassert/assert.go#L49) here

Reply:
Done.